admin/server/reports.go

```diff
@@ -488,6 +487,11 @@ func (s *Server) yamlForManagedReport(opts *adminv1.ReportOptions, ownerUserID s
 	res.Annotations.AdminNonce = time.Now().Format(time.RFC3339Nano)
 	res.Annotations.WebOpenPath = opts.WebOpenPath
 	res.Annotations.WebOpenState = opts.WebOpenState
+	if opts.Explore != "" && opts.Canvas != "" {
+		return nil, fmt.Errorf("cannot set both explore and canvas")
+	}
+	res.Annotations.Explore = opts.Explore
+	res.Annotations.Canvas = opts.Canvas
```

Comment:
Should we check somewhere if the report has an explore or canvas? And if not, then skip generating the open link (since it would not actually work).

Some legacy reports and YAML-managed reports may not have an explore or canvas.